build/genesis_test.go

```diff
@@ -0,0 +1,39 @@
+package build_test
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/lotus/build"
+)
+
+func TestGenesis(t *testing.T) {
+	for _, test := range []struct {
+		path string
+	}{
+		{path: "genesis/butterflynet.car.zst"},
+		{path: "genesis/calibnet.car.zst"},
+		{path: "genesis/interopnet.car.zst"},
+		{path: "genesis/mainnet.car.zst"},
+	} {
+		t.Run(test.path, func(t *testing.T) {
+			subject, err := os.Open(test.path)
+			require.NoError(t, err)
+
+			gotIsCompressed, err := build.IsZstdCompressed(subject)
+			require.NoError(t, err)
+			require.True(t, gotIsCompressed)
+
+			gotDecompressed, err := build.DecompressAsZstd(subject)
+			require.NoError(t, err)
+			require.NotEmpty(t, gotDecompressed)
+
+			gotIsCompressed, err = build.IsZstdCompressed(bytes.NewReader(gotDecompressed))
+			require.NoError(t, err)
+			require.False(t, gotIsCompressed)
+		})
+	}
```

Comment:
how about a negative case, `build.IsZsrdCompressed("genesis_test.go")`

Reply:
As In not zstd and not CAR? 

Happy to add though it seems superfluous considering the assertions that check uncompressed genesis files are not zstd compressed.